### PR TITLE
Update MQTT shim to use the coreMQTT from submodule

### PR DESCRIPTION
**Pull request by leegeth**
Update MQTT shim to use the coreMQTT from submodule.

Description
-----------
Update MQTT shim to use the coreMQTT from submodule. These changes include
1. Update CMakeLists.txt of MQTT shim to add coreMQTT as a dependency.
2. Update the API calls that got updated.
3. Create a CMakeLists.txt for building coreMQTT.
4. Remove the redundant mqtt_v2 folder.
5. Move core_mqtt_config.h to demos/tests.

Checklist:
----------
- [x ] I have tested my changes. No regression in existing tests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk_v4/standard/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
+afr_module(NAME "coreMQTT" INTERFACE)
```
Comment (aggarw13) [inline libraries/c_sdk_v4/standard/CMakeLists.txt:1]: For OCW visibility, the module cannot be an `INTERFACE`

You can use my branch for MQTT integration test porting as reference :) 
https://github.com/aggarw13/amazon-freertos/blob/mqtt/integration-tests-port/libraries/c_sdk_v4/standard/CMakeLists.txt

Comment (leegeth) [reply libraries/c_sdk_v4/standard/CMakeLists.txt:1]: Thanks. That's a good suggestion.

File: libraries/c_sdk/standard/mqtt/include/core_mqtt_config.h [hunk 2]
```diff
@@ -1,5 +1,5 @@
-#ifndef MQTT_CONFIG_H_
-#define MQTT_CONFIG_H_
+#ifndef CORE_MQTT_CONFIG_H_
```
Comment (aggarw13) [inline libraries/c_sdk/standard/mqtt/include/core_mqtt_config.h:1]: Can the config file be removed? We are keeping the config file only in the tests and demo projects. 
In this [PR](https://github.com/aws/amazon-freertos/pull/2482) for adding coreMQTT integration tests, I have added the config file to both the `aws_test` and `aws_demo` folders of the target board(s)

Comment (leegeth) [reply libraries/c_sdk/standard/mqtt/include/core_mqtt_config.h:1]: Yea. I was thinking to create a separate PR for doing that. I knew that by removing interface from cmake, we will have to create core_mqtt_config.h in each demo/tests for vendor.

File: libraries/c_sdk_v4/standard/CMakeLists.txt (hunk 1 above)
Comment (leegeth) [reply libraries/c_sdk_v4/standard/CMakeLists.txt]: Changed the plan to update this PR itself. Addressed it

File: libraries/c_sdk/standard/mqtt/include/core_mqtt_config.h (hunk 2 above)
Comment (leegeth) [reply libraries/c_sdk/standard/mqtt/include/core_mqtt_config.h]: Addressed the comments.

**Review by aggarw13 — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**